content/en/observability_pipelines/guide/get_started_with_the_custom_processor.md

```diff
@@ -0,0 +1,287 @@
+```
+
+
+Alternatively, you can rewrite the original `payload` value with the decoded value by replacing `decoded_payload` in the previous function with `payload`.
+
+```
+.payload = decode_base64!(.payload)
+```
+
+#### Output
+
+The output when you use `decoded_payload` to store the decoded value.
+
+```
+{
+    "timestamp": "2025-05-28T19:30:00Z",
+    "level": "info",
+    "source": "auth-service",
+    "user_id": "user_9876",
+    "payload": "VXNlciByZXF1ZXN0ZWQgYWNjZXNzIHRvIHByb3RlY3RlZCByZXNvdXJjZQ==",
+    "decoded_payload": "User requested access to protected resource"
+}
+```
+
+## Decode an entire Base64-encoded event
+
+### Example
+
+#### Input
+
+Example input of an event encoded in base64:
+
+```
+{
+    "raw": "eyJ0aW1lc3RhbXAiOiAiMjAyNS0wNS0yOFQxOTozMDowMFoiLCAibGV2ZWwiOiAiaW5mbyIsICJtessagemIjogIlVzZXIgbG9naW4gc3VjY2Vzc2Z1bCJ9"
+}
+```
+
+#### Custom function
+
+The script to decode the entire Base64-encoded event `raw`.
+
+```
+.json_string = decode_base64!(.raw)`
+.full_event = parse_json!(.json_string)
+. = .full_event
+```
+
+**Note:** The syntax `. = .full_event` is shorthand for replacing the entire event with the contents of a field.
+
+#### Output
+
+```
+{
+    "timestamp": "2025-05-28T19:30:00Z",
+    "level": "info",
+    "message": "User login successful"
+}
+```
+
+## Encode Base64
+
+For outgoing log fields or events that you want to encode in Base64, use the [`encode_base64`][2] function to encode the field or event. This function's syntax also works for [`encode_base16`][3].
+
+### Example
+
+#### Input
+
+Example log event containing the `message` field that you want to encode in Base64:
+
+```
+{
+    "timestamp": "2025-05-28T19:30:00Z",
+    "level": "info",
+    "source": "auth-service",
+    "user_id": "user_9876",
+    "message": "User login successful"
+}
+```
+
+#### Custom function
+
+Use the `encode_base64` function to decode `message` and store the result in a new field called `encoded_message`.
+
+```
+.encoded_message = encode_base64!(.message)
+```
+
+Alternatively, you can overwrite the original message field (`message`) with the decoded value by replacing `encoded_message` in the previous function with `message`.
+
+```
+.message = encode_base64!(.message)
+```
+
+#### Output
+
+The output when you use `encoded_message` to store the encoded value.
+
+```
+{
+    "timestamp": "2025-05-28T19:30:00Z",
+    "level": "info",
+    "source": "auth-service",
+    "message": "User login successful",
+    "encoded_message": "VXNlciBsb2dpbiBzdWNjZXNzZnVs"
+}
+```
+
+## Remap timestamps for historical logs
+
+If you want to migrate archived logs from other platforms, ensuring tthosehe logs have the correct historical timestamp is essential. Refer to [Migrate Historical Logs][4] blog on how to migrate historical logs from specific platforms like Splunk and Elasticsearch. Remapping logs with historical timestamps enables you to handle older logs stored for compliance, audit, and archive purposes.
```

Comment:
```suggestion
If you want to migrate archived logs from other platforms, ensuring those logs have the correct historical timestamp is essential. Refer to the [Migrate Historical Logs][4] blog post on how to migrate historical logs from specific platforms like Splunk and Elasticsearch. Remapping logs with historical timestamps enables you to handle older logs stored for compliance, audit, and archive purposes.
```

adding "post" to "blog," so that people don't think there's a whole publication called Migrate Historical Logs